### ISSUE TITLE
[1.20.1 Bug]: 

### ISSUE THREAD

**Issue by sszhuanxiang** [DragonSurvivalTeam/DragonSurvival#568]
### ■ Dragon Survival version

DragonSurvival-1.20.1-09.05.2024_ALPHA

### ■ Forge/Neoforge version

forge-47.3.0

### ■ Single game or server?

Local network (LAN, Hamachi, etc.)

### ■ Describe the bug

When I run the game and try to load a save file, it hangs on “loading data“. It seems no response. After checking the error log, I found there seems to be an issue with accessing a URL. Here are the relevant URL details and screenshots. It seems my network access is being blocked:

[14:55:15] [Render thread/WARN] [Dragon Survival/]: Reader could not be closed
java.io.IOException: Server returned HTTP response code: 403 for URL: https://web-api.gitcode.com/api/v1/projects/mirrors%2FDragonSurvivalTeam%2FDragonSurvival/repository/tree?ref=master&path=src/test/resources&per_page=100&page=1
[Log4.txt](https://github.com/user-attachments/files/16101566/Log4.txt)
![1720112736153](https://github.com/DragonSurvivalTeam/DragonSurvival/assets/136944164/7ab1f4b7-ef49-46da-a8da-0d0ab58a864b)

### ■ Crash Report File and Logs

[Log4.txt](https://github.com/user-attachments/files/16101568/Log4.txt)

**Comment by sszhuanxiang** [comment]
Of course, two players who connected to my LAN are also experiencing the same issue.

**Comment by Psither** [comment]
It appears that Github has changed their handling for HTML requests coming from China, see: 1195160

**Comment by Eirin-lin** [comment]
This bug is also found on 1.19.2. To address this issue temporarily, you can use proxy software like Clash to reject all connections to `web-api.gitcode.com`.
```
rules:
  - DOMAIN-SUFFIX,web-api.gitcode.com,REJECT
```

**Comment by Inhrtbury2998** [comment]
I think it's because the gitcode site has been under maintenance for the last few days, so it's going to 403.See [this](https://gitcode.com/Gitcode-offical-team/GitCode-Docs/issues/1559)
